### PR TITLE
mine_charity fix

### PR DESCRIPTION
**Pull request by Gnaget**
fixed mine_charity for the new default scheduler and the alt slice scheduler

### CODE REVIEW

**Comment by c00w** [comment]
Can you change all the references to friendly in function and variable names to charity please? And public Domain or transfer copyright to me?

**Comment by Gnaget** [comment]
done, and I added hashrate to the userdata based on a rolling 5 minute share submit.  All code is public domain, and all future changes in this fork will also be public domain if you want to simply take that as a global answer

**Comment by c00w** [comment]
Why did you remove deepbit? Can you please put it back? I'm literally fighting with my LP based deepbit hopping as we speak.

**Comment by Gnaget** [comment]
Can you just ignore the changes in pools.cfg.  I didn't intend to include those, I do a lot of custom stuff in there, and I didn't notice it was included.  Personally, I'm not sure why that config was added, makes updating a pain in the ass now

**Comment by c00w** [comment]
Um if I ignore changes in pool.cfg then kiwi pool won't work. Pools.cfg was added so we could add pools and update api methods without forcing everyone to change their personnel configuration to take advantage of it.

**Comment by Gnaget** [comment]
That is great in theory, can pools only be in user.cfg?  Ignore Kiwi as well, it isn't working yet.  That is the point, if I want a add a pool and don't want to check it in, can we just put it in user.cfg?

**Comment by c00w** [comment]
Yep.

**Comment by Gnaget** [comment]
Replaced my pools.cfg with yours

**Comment by Gnaget** [comment]
Well, I thought I did..  let me try again

**Comment by Gnaget** [comment]
ok, it is good now, have at it
